File: Exercism/typescript/difference-of-squares/difference-of-squares.ts
export class Squares {

    public squareOfSum: number;
    public sumOfSquares: number;
    public difference: number;

    constructor(x: number) {
        const arr = Array.from(Array(x+1).keys());
        this.squareOfSum = Math.pow(arr.reduce((sum, current) => sum + current, 0), 2);
        this.sumOfSquares = arr.reduce((sum, current) => sum + Math.pow(current, 2), 0);
        this.difference = this.squareOfSum - this.sumOfSquares;
    }
}

export default Squares

File: Exercism/go/difference-of-squares/difference_of_squares.go
package diffsquares

import "math"

func SquareOfSum(x int) int {
	var result int

	for i := 1; i <= x; i++ {
		result += i
	}

	return int(math.Pow(float64(result), 2))
}

func SumOfSquares(x int) int {
	var result int

	for i := 1; i <= x; i++ {
		result += int(math.Pow(float64(i), 2))
	}

	return result
}

func Difference(x int) int {
	return SquareOfSum(x) - SumOfSquares(x)
}
